plutus-tx-plugin/src/PlutusTx/Compiler/Expr.hs

```diff
@@ -765,11 +764,17 @@ compileExpr e = traceCompilation 2 ("Compiling expr:" GHC.<+> GHC.ppr e) $ do
       case maybeDef of
         Just term -> pure term
         Nothing ->
-          throwSd FreeVariableError $
-            "Variable"
-              GHC.<+> GHC.ppr n
-              GHC.$+$ (GHC.ppr $ GHC.idDetails n)
-              GHC.$+$ (GHC.ppr $ GHC.realIdUnfolding n)
+          -- No other cases apply; compile the unfolding of the var
```

Comment:
I would mildly prefer to keep this as a separate case and put it later. That way the fallthrough case is more obviously part of the series of `Var`-related matches. We can put the necessary checks into the pattern here:
```
GHC.Var n | let lexName = LexName $ GHC.getName n, Just term <- PIR.lookupTerm annMayInline lexName = ...
```

Reply:
`lookupTerm` is a monadic action and can't be put in the guards!

Reply:
bah